internal/controller/instaslice_controller.go

```diff
@@ -105,61 +119,86 @@ func (r *InstasliceReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 	}
 
-	var instasliceList inferencev1alpha1.InstasliceList
-
-	if err := r.List(ctx, &instasliceList, &client.ListOptions{}); err != nil {
-		log.FromContext(ctx).Error(err, "Error listing Instaslice")
-	}
-
 	// failed pods are not deleted by InstaSlice, finalizer is removed so that user can
-	// delete the pod and later InstaSlice will clean up the MIG slice.
+	// delete the pod.
 	if pod.Status.Phase == v1.PodFailed && controllerutil.ContainsFinalizer(pod, finalizerName) {
+		allocationNotFound := true
 		for _, instaslice := range instasliceList.Items {
 			for _, allocation := range instaslice.Spec.Allocations {
-				found, result, err := r.processInstasliceAllocation(ctx, instaslice.Name, string(pod.UID), allocation, req)
-				if err != nil {
-					return result, err
-				}
-				if found {
-					return ctrl.Result{}, nil
+				if pod.UID == types.UID(allocation.PodUUID) {
+					allocationNotFound = false
+					if allocation.Allocationstatus == inferencev1alpha1.AllocationStatusCreating {
+						return ctrl.Result{RequeueAfter: 2 * time.Second}, nil
+					}
+					if allocation.Allocationstatus == inferencev1alpha1.AllocationStatusCreated || allocation.Allocationstatus == inferencev1alpha1.AllocationStatusUngated {
+						resultDeleting, errInDeleting := r.setInstasliceAllocationToDeleting(ctx, instaslice.Name, string(pod.UID), allocation)
+						if errInDeleting != nil {
+							return resultDeleting, nil
```

Comment:
hmm. errInDeleting is not returned... It's ok in golang to shadow the variable `err`. Instead of creating errInDeleting or errInRemove, I suggest using just `err`.

Reply:
Thanks, Done